src/sentry/relay/config/metric_extraction.py

```diff
@@ -409,42 +409,56 @@ def convert_widget_query_to_metric(
     if not widget_query.aggregates:
         return metrics_specs
 
-    for aggregate in widget_query.aggregates:
-        metrics.incr(
-            "on_demand_metrics.before_widget_spec_generation",
-            tags={"prefilling": prefilling},
+    aggregates = widget_query.aggregates or []
+    groupbys = widget_query.columns
```

Comment:
As seen in line 426:
<img width="219" alt="image" src="https://github.com/getsentry/sentry/assets/44410/efc75db5-9dff-4ef7-b667-bffbb1703637">